app/models/services/commitment_props.rb

```diff
@@ -0,0 +1,322 @@
+                },
+                optionsCaption: '-Select a country-'
+              },
+              {
+                type: 'expression',
+                name: 'latlongHeading',
+                title: 'Please provide the latitude and longitude of the area'
+              },
+              {
+                type: 'text',
+                name: 'latitude',
+                title: 'Latitude',
+                titleLocation: 'left',
+                hideNumber: true
+              },
+              {
+                type: 'text',
+                name: 'longitude',
+                title: 'Longitude',
+                titleLocation: 'left',
+                hideNumber: true
+              },
+              {
+                type: 'file',
+                name: 'geospatialFilename',
+                title:
+                  'If you have spatial data you would like to share, please upload it here.',
+                description: '(Optional field)',
+                hideNumber: true,
+                allowImagesPreview: false,
+                maxSize: 26_214_400
+              },
+              {
+                type: 'text',
+                name: 'areaHa',
+                title: 'How many hectares does/will the area cover?'
+              }
+            ],
+            title: 'Location'
+          },
+          {
+            name: 'durationPage',
+            elements: [
+              {
+                type: 'dropdown',
+                name: 'committedYear',
+                title: 'Please state the year the commitment was made',
+                # defaultValue: [],
+                choices: %w[
+                  2022
+                  2021
+                  2020
+                  2019
+                  2018
+                  2017
+                  2016
+                  2015
+                  2014
+                  2013
+                  2012
+                ],
+                optionsCaption: '-Select-'
+              },
+              {
+                type: 'dropdown',
+                name: 'duration',
+                title:
+                  'How many years do you expect the commitment to be in place for?',
+                  # defaultValue: [],
+                choices: %w[
+                  5
+                  6
+                  7
+                  8
+                  9
+                  10
+                  11
+                  12
+                  13
+                  14
+                  15
+                ],
+                optionsCaption: '-Select-'
+              }
+            ],
+            title: 'Duration'
+          },
+          {
+            name: 'stagePage',
+            elements: [
+              {
+                type: 'radiogroup',
+                name: 'stage',
+                title: 'Please select the stage of implementation of your commitment',
+                # defaultValue: [],
+                choices: ['Committed', 'In progress', 'Implemented']
+              },
```

Comment:
when 31 gets merged there will be a constant in the the Commitment class, used to validate that the stage submitted (mostly to handle dodgy data in a csv upload): ```STAGE_OPTIONS = ['In progress', 'Committed', 'Implemented']```

and you will be able to access them here with ```Commitment::STAGE_OPTIONS```

Reply:
👍 I'll Add all TODOs to the ticket